### PR TITLE
[RooFit] RooGenericPdf default Constructor

### PR DESCRIPTION
**Pull request by amarini**
adding back default constructor for RooGenericPdf (for I/O)

### CODE REVIEW

File: roofit/roofitcore/inc/RooGenericPdf.h [hunk 1]
```diff
@@ -25,6 +25,7 @@ class RooArgList ;
 class RooGenericPdf : public RooAbsPdf {
 public:
   // Constructors, assignment etc
+  inline RooGenericPdf(){}
```
Comment (pcanal) [inline roofit/roofitcore/inc/RooGenericPdf.h]: maybe:
```suggestion
  RooGenericPdf() = default;
```
[Note that the inline keyword is already implied if the definition is inside the class declaration]

**Comment by hageboeck** [comment]
Hello @amarini,

could you elaborate a bit what the I/O problem is? All tests were green so far.

Also, a question to either @pcanal, @linev, @vgvassilev (I don't remember who raised this issue):
There was a discussion about problems when `= default`ing a constructor. Is that still an issue or can we merge this?

**Comment by linev** [comment]
Hi,

It is an issue for all TObject-derived classes:

https://sft.its.cern.ch/jira/browse/ROOT-10300

One cannot use ` = default` for constructor

**Comment by amarini** [comment]
@hageboeck,  an constructor w/o arguments (or a more complicate one) is needed to do I/O, for example save a RooWorkspace to file and load it later.
I verified the first implementation: 
~~~c++
inline RooGenericPdf(){}
~~~
I assumed that the =default was behaving the same but looking at the thread it seems not to. 

I reverted the =default to the previous stage.

**Comment by hageboeck** [comment]
Ok, whether it's inline or not is really not a problem. Thanks for the pull request.

**Pull request merged**